sdk/storage/azure_storage_blob/src/parsers.rs

```diff
@@ -35,3 +40,51 @@ pub fn format_page_range(offset: u64, length: u64) -> Result<String, Error> {
     let content_range = format!("bytes={}-{}", offset, end_range);
     Ok(content_range)
 }
+
+/// Takes in a HashMap of blob tags and serializes them into the `BlobTags` model.
+///
+/// # Arguments
+///
+/// * `tags` - A hash map containing the name-value pairs associated with the blob as tags.
+pub(crate) fn serialize_blob_tags(tags: HashMap<String, String>) -> BlobTags {
+    let sorted_tags: BTreeMap<_, _> = tags.into_iter().collect();
+    let blob_tags = sorted_tags
+        .into_iter()
+        .map(|(k, v)| BlobTag {
+            key: Some(k),
+            value: Some(v),
+        })
+        .collect();
+    BlobTags {
+        blob_tag_set: Some(blob_tags),
+    }
+}
+
+/// Takes in a `get_tags()` response and deserializes the `BlobTags` model into a HashMap of blob tags.
+///
+/// # Arguments
+///
+/// * `response` - The `get_tags()` response to be deserialized.
+pub(crate) async fn deserialize_blob_tags(
+    response: Response<BlobTags, XmlFormat>,
+) -> azure_core::Result<Response<HashMap<String, String>, JsonFormat>>
+where
+{
+    let mut blob_tags_map: HashMap<String, String> = HashMap::new();
+    let status = response.status();
+    let headers = response.headers().clone();
+    let blob_tags = response.into_body().await?;
+
+    if let Some(blob_tag_set) = blob_tags.blob_tag_set {
+        for tag in blob_tag_set {
+            if let (Some(k), Some(v)) = (tag.key, tag.value) {
+                blob_tags_map.insert(k, v);
+            }
+        }
+    }
+
+    let request_content: RequestContent<HashMap<String, String>> =
+        RequestContent::try_from(blob_tags_map)?;
+    let raw_response = RawResponse::from_bytes(status, headers, request_content.body());
+    Ok(raw_response.into())
```

Comment:
This works, but is still a little clunky. The generated code returns a `Response<BlobTags, XmlFormat>`.  We are using `RequestContent::try_from(blob_tags_map)?;` which is implemented as returning `RequestContent<T, F=JsonFormat>`. Therefore, when I use the `.body()` to pass to the repacked `RawResponse`, it will be `Json` and so we need to change our deserializer (and thus response object) to be a:
`Result<Response<HashMap<String, String>, JsonFormat>>`

Reply:
@heaths, I would argue that Core should have XML support. We can't be the only customer in all of Azure that uses XML and even if that's true, it feels like something that should be part of a library like Core.

Reply:
As discussed offline (again 😆) it is impossible for XML to be generically implemented because of the root tag.